File: backend/src/main/java/com/urbanfresh/service/AuthService.java
package com.urbanfresh.service;

import com.urbanfresh.dto.request.LoginRequest;
import com.urbanfresh.dto.request.RegisterRequest;
import com.urbanfresh.dto.response.LoginResponse;
import com.urbanfresh.dto.response.RegisterResponse;

/**
 * Service Layer – Contract for authentication-related operations.
 * Keeps the controller decoupled from implementation details.
 */
public interface AuthService {

    /**
     * Register a new customer account.
     *
     * @param request validated registration data
     * @return response with the created user's public info
     * @throws com.urbanfresh.exception.DuplicateEmailException if email is taken
     */
    RegisterResponse registerCustomer(RegisterRequest request);

    /**
     * Authenticate a user and issue a JWT token.
     *
     * @param request validated login credentials (email + password)
     * @return response with JWT token and user info
     * @throws com.urbanfresh.exception.InvalidCredentialsException if credentials are wrong
     */
    LoginResponse login(LoginRequest request);
}

File: frontend/src/services/authService.js
/**
 * Service Layer – Axios calls for authentication endpoints.
 * Centralizes all auth API interactions; keeps components free of HTTP logic.
 */
import axios from 'axios';

const API_BASE = 'http://localhost:8080/api/auth';

/**
 * Register a new customer account.
 * @param {object} data - { name, email, password, phone }
 * @returns {Promise} resolved with RegisterResponse on success
 * @throws axios error with response.data containing field errors or message
 */
export const registerCustomer = (data) =>
  axios.post(`${API_BASE}/register`, data);

/**
 * Authenticate a user and receive a JWT token.
 * @param {object} data - { email, password }
 * @returns {Promise} resolved with { token, email, name, role, message }
 * @throws axios error with response.data containing error message (401/400)
 */
export const loginUser = (data) =>
  axios.post(`${API_BASE}/login`, data);

